src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  static checkDead() {
+    Animal.alive = Animal.alive.filter((animal) => {
+      return animal.health > 0;
+    });
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = false;
```

Comment:
The `hidden` parameter in the `Herbivore` constructor is not used, and the `hidden` property is always set to `false`. If the intention is to allow setting the initial hidden state, you should use the `hidden` parameter to initialize the `hidden` property.